### PR TITLE
defensive guard checking

### PR DESCRIPTION
**Pull request by InoMurko**

### CODE REVIEW

File: lib/ex_rlp/encode.ex [hunk 1]
```diff
@@ -23,7 +23,7 @@ defimpl ExRLP.Encode, for: BitString do
     <<prefix>> <> item
   end
 
-  defp encode_item(item) when is_binary(item) do
+  defp encode_item(item) do
```
Comment (ayrat555) [inline lib/ex_rlp/encode.ex]: why did you remove guards? What is the benefit?

Comment (InoMurko) [reply lib/ex_rlp/encode.ex]: @ayrat555 the benefit is performance. Second, it doesn’t make sense to check if the argument is binary in you’re implementing the binary protocol, same applies for integer protocol. It’s defensive.
